proof/invariant-abstract/AARCH64/ArchDetype_AI.thy

```diff
@@ -636,8 +636,8 @@ interpretation Detype_AI_2
         Detype_AI_2.intro
         by blast
 
-context begin interpretation Arch .
-lemma delete_objects_invs[wp]:
+(* generic consequence of architecture-specific details *)
+lemma (in Arch) delete_objects_invs[wp]:
   "\<lbrace>(\<lambda>s. \<exists>slot. cte_wp_at ((=) (cap.UntypedCap dev ptr bits f)) slot s
```

Comment:
This one is less weird than I originally thought. It's in an Arch file and usually would be inside the big `context Arch` block that spans the entire file, but here it needed to be after the `interpretation` part that comes above. So it's mostly a "usual" Arch lemma that is getting exported and requalified.

Looks all good to me, and the new handling of the `wp` attribute is better.

Reply:
The issue here is that the qualified name of `delete_objects_invs` becomes `Arch.delete_objects_invs`, instead of what we normally would do via global_prefix (`AARCH64.delete_objects_invs`). In this case that should be ok, because it's immediately requalified into the theory context (and also tagged `[wp]` for that context), but it's still a bit of a finicky policy to try canonise.

Reply:
Got it. We don't seem to have many of these, and for exceptional cases I think this is Ok.

Reply:
Would the 'canonical' approach be to export enough facts so that this can be proved in a generic file outside of `Arch`?

Reply:
Yeah, I think exporting enough facts would be the right idea. I think someone took a shortcut here, but I'm not clear on why exactly.